OptimoveSDK/optimove-sdk/src/main/java/com/optimove/android/optimobile/AnalyticsUploadHelper.java

```diff
@@ -44,44 +45,30 @@ enum Result {
                 events = eventsResult.first;
                 maxEventId = eventsResult.second;
             }
-        }
-        catch (SQLiteException e) {
+        } catch (SQLiteException e) {
             e.printStackTrace();
             return Result.FAILED_RETRY_LATER;
         }
+        catch (Optimobile.PartialInitialisationException e) {
+            return Result.FAILED_NO_RETRY;
+        }
 
         return Result.SUCCESS;
     }
 
-    private boolean flushBatchToNetwork(Context context, ArrayList<JSONObject> events, long maxEventId) {
+    private boolean flushBatchToNetwork(Context context, ArrayList<JSONObject> events, long maxEventId) throws Optimobile.PartialInitialisationException {
         // Pack into JSON
         JSONArray data = new JSONArray(events);
-        String dataStr = data.toString();
-        if (null == dataStr) {
-            return false;
```

Comment:
behaviour change. Before if `toString` was giving null we would retry. Now it will fail.

1. not clear why this check was necessary at all -- `ArrayList<JSONObject> events` should be jsonarray-able
2. if it was failing -- we would silently never send event?